apps/meteor/client/views/omnichannel/currentChats/CurrentChatsRoute.tsx

```diff
@@ -105,75 +123,43 @@ const CurrentChatsRoute = (): ReactElement => {
 		department: '',
 		from: '',
 		to: '',
-		customFields: {},
 		current: 0,
 		itemsPerPage: 25 as 25 | 50 | 100,
 		tags: [] as string[],
 	});
-	const [sort, setSort] = useState<[string, 'asc' | 'desc' | undefined]>(['ts', 'desc']);
-
-	const debouncedParams = useDebouncedValue(params, 500);
-	const debouncedSort = useDebouncedValue(sort, 500);
-	const query = useQuery(debouncedParams, debouncedSort);
+	const t = useTranslation();
+	const id = useRouteParameter('id');
 
 	const onHeaderClick = useMutableCallback((id) => {
-		const [sortBy, sortDirection] = sort;
-
 		if (sortBy === id) {
-			setSort([id, sortDirection === 'asc' ? 'desc' : 'asc']);
+			setSort(id, sortDirection === 'asc' ? 'desc' : 'asc');
 			return;
 		}
-		setSort([id, 'asc']);
+		setSort(id, 'asc');
 	});
 
+	const debouncedParams = useDebouncedValue(params, 500);
+	const debouncedCustomFields = useDebouncedValue(customFields, 500);
+
+	const debouncedSort = useDebouncedValue(
+		useMemo(() => [sortBy, sortDirection], [sortBy, sortDirection]),
+		500,
+	) as ['fname' | 'departmentId' | 'servedBy' | 'ts' | 'lm' | 'open', 'asc' | 'desc'];
```

Comment:
I've seen this type more than once. Maybe it would be a good ideia to give it a name and reuse it throughout the code.

Reply:
Where exactly have you seen them? Could be a variation of this indeed